include/kframework/evm.md

```diff
@@ -1385,6 +1429,33 @@ The various `CALL*` (and other inter-contract control flow) operations will be d
 
     rule #computeValidJumpDestsWithinBound(PGM, I, RESULT) => #computeValidJumpDests(PGM, I +Int 1, RESULT ListItem(I)) requires PGM [ I ] ==Int 91
     rule #computeValidJumpDestsWithinBound(PGM, I, RESULT) => #computeValidJumpDests(PGM, I +Int #widthOpCode(PGM [ I ]), RESULT) requires notBool PGM [ I ] ==Int 91
+
+   syntax Map ::= #computeBasicBlocks(Bytes)                [function, memo]
+                | #computeBasicBlocks(Bytes, Int, Int, Map) [function, klabel(#computeBasicBlocksAux)]
+ // -------------------------------------------------------------------------------------------------------
+   rule #computeBasicBlocks(PGM) => #let BLOCKS = #computeBasicBlocks(PGM, 0, 0, .Map)
+                                    #in #if size(BLOCKS) ==Int 0 #then (0 |-> PGM) #else BLOCKS #fi
+
+   syntax Map ::= #computeBasicBlocksWithinBound(Bytes, Int, Int, Map) [function]
+ // -----------------------------------------------------------------------------------
+   rule #computeBasicBlocks(PGM, I, J, RESULT) => RESULT (J |-> substrBytes(PGM, J, lengthBytes(PGM)))
+     requires I >=Int lengthBytes(PGM) andBool J >Int 0
+   rule #computeBasicBlocks(PGM, I, J, RESULT) => RESULT
+     requires I >=Int lengthBytes(PGM) andBool J ==Int 0
+   rule #computeBasicBlocks(PGM, I, J, RESULT) => #computeBasicBlocksWithinBound(PGM, I, J, RESULT)
+     requires I <Int  lengthBytes(PGM)
+
+   rule #computeBasicBlocksWithinBound(PGM, I, _, RESULT) =>
+        #computeBasicBlocks(PGM, I +Int 1, I, RESULT)
+     requires PGM [ I ] ==Int 91
+      andBool I ==Int 0
+   rule #computeBasicBlocksWithinBound(PGM, I, J, RESULT) =>
+        #computeBasicBlocks(PGM, I +Int 1, I, RESULT (J |-> substrBytes(PGM, J, I)))
+     requires PGM [ I ] ==Int 91
+      andBool I >Int 0
+   rule #computeBasicBlocksWithinBound(PGM, I, J, RESULT) =>
+        #computeBasicBlocks(PGM, I +Int #widthOpCode(PGM [ I ]), J, RESULT)
+     requires notBool PGM [ I ] ==Int 91
```

Comment:
I guess `computeValidJumpDests` could just re-use `computeBasicBlocks` then, by inspecting each map element and keeping the key if the first item of the basic block is a `JUMPDEST`?

Or, do we really need `computeValidJumpDests` anymore? Right now it's used here: https://github.com/runtimeverification/evm-semantics/blob/master/include/kframework/evm.md#evm-control-flow

```
    rule <k> JUMP DEST => #endBasicBlock... </k>
         <pc> _ => DEST </pc>
         <jumpDests> DESTS </jumpDests>
      requires DEST in DESTS

    rule <k> JUMP DEST => #end EVMC_BAD_JUMP_DESTINATION ... </k>
         <jumpDests> DESTS </jumpDests>
      requires notBool DEST in DESTS
```

But these uses could be pretty easily replaced with:

```
    rule <k> JUMP DEST => #endBasicBlock... </k>
         <schedule> SCHED </schedule>
         <pc> _ => DEST </pc>
         <basic-blocks> ... DEST |-> BA ... </jumpDests>
      requires #unconsOpCode(SCHED, BA) ==K JUMPDEST

    rule <k> JUMP DEST => #end EVMC_BAD_JUMP_DESTINATION ... </k> [owise]
```

Reply:
It indeed does not look like we need `<jumpDest>` anymore. I think we can use a even simpler side condition:

```
    rule <k> JUMP DEST => #endBasicBlock... </k>
         <schedule> SCHED </schedule>
         <pc> _ => DEST </pc>
         <basic-blocks> BLOCKS </jumpDests>
      requires DEST in_keys(BLOCKS)

    rule <k> JUMP DEST => #end EVMC_BAD_JUMP_DESTINATION ... </k> [owise]
```
since the the keys in the `BLOCKS` map are all JUMPDEST (and 0 if the first opcode is not JUMPDEST)